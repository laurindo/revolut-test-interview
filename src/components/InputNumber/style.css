input.input-number {
  font-size: 2em;
  border-radius: 5px;
  color: #0e5fb7;
  padding: 0 10px;
  border: solid 1px #e5e7e8;
  padding: 4px 12px 4px;
  text-align: right;
  width: 100%;
}
