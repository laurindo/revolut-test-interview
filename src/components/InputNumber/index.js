import React, { Component } from "react";
import './style.css';

const InputNumber = ({ value, readonly = false, onChange }) => (
  <input
    className="input-number"
    readOnly={readonly}
    type="text"
    value={value}
    onChange={(evt) => onChange(evt.target.value)} />
);

export default InputNumber;
