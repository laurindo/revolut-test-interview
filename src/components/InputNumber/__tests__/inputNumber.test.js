import React from 'react';
import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import InputNumber from '../index';

configure({ adapter: new Adapter() });

describe('InputNumber Component', () => {
  test('should render the component', () => {
    const component = shallow(<InputNumber value="2" onChange={() => { }} />);
    expect(component).toMatchSnapshot();
    expect(component.find('input')).toBeDefined();
    expect(component.find('input').get(0).props.className).toEqual('input-number');
    expect(component.find('input').get(0).props.readOnly).toEqual(false);
    expect(component.find('input').get(0).props.type).toEqual('text');
    expect(component.find('input').get(0).props.value).toEqual('2');
  });
});
