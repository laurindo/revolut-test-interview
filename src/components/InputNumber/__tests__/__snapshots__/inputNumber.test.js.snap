exports[`InputNumber Component should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputNumber
    onChange={[Function]}
    value="2" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "input-number",
      "onChange": [Function],
      "placeholder": "0",
      "readOnly": false,
      "type": "text",
      "value": "2",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "input-number",
        "onChange": [Function],
        "placeholder": "0",
        "readOnly": false,
        "type": "text",
        "value": "2",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
